openwisp_radius/api/serializers.py

```diff
@@ -350,6 +350,10 @@ def save(self, request):
         adapter = get_adapter()
         user = adapter.new_user(request)
         self.cleaned_data = self.get_cleaned_data()
+        # ensuring email is always converted to lowercase
+        email = str(user.email)
+        email = email.lower()
+        user.email = email
```

Comment:
Instead of doing this here, we can override the default `get_cleaned_data` method and make email lowercase in that method. IMO, it will keep code cleaner.

Reply:
correct, @sankha555 please use field level validation, read here:
https://www.django-rest-framework.org/api-guide/serializers/#field-level-validation